react/src/Sidebar/NestedSidebarItem.tsx

```diff
@@ -0,0 +1,53 @@
+import { FC, PropsWithChildren, useMemo } from 'react'
+import {
+  Accordion,
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  AccordionProps,
+  Flex,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { merge } from 'lodash'
+
+import { useSidebarStyles } from './SidebarStylesContext'
+import { BaseSidebarItem } from './types'
+
+export interface NestedSidebarItemProps extends BaseSidebarItem {
+  /**
+   * Whether the item is the root item of the sidebar. Determines styling.
+   */
+  root?: boolean
+  props?: AccordionProps
+}
+
+export const NestedSidebarItem: FC<
+  PropsWithChildren<NestedSidebarItemProps>
+> = ({ icon, label, props, root, children }) => {
+  const styles = useSidebarStyles()
+
+  const itemStyles = useMemo(() => {
+    return merge({}, styles.parent, root ? {} : styles.child, styles.item)
+  }, [root, styles.child, styles.item, styles.parent])
+
+  return (
+    <Accordion variant="sidebar" colorScheme="main" allowToggle {...props}>
+      <AccordionItem>
+        <AccordionButton sx={itemStyles}>
+          <Flex gap="0.5rem">
+            {icon}
+            <Text>{label}</Text>
+          </Flex>
+          <AccordionIcon ml="0.5rem" />
+        </AccordionButton>
```

Comment:
I noticed that the `itemStyles` passed to `AccordionButton` don't affect its children as expected. E.g. adding `textStyle` won't actually change the inner `<Text>{label}</Text>`

Reply:
Thx for catch. Will update

Reply:
Ah, fixed in #234 haha. Should probably have collapsed the two branches